gpu_support/add_gpu_support.sh

```diff
@@ -0,0 +1,192 @@
+latest_cuda_compat_url="$($(dirname "$BASH_SOURCE")/get_latest_cuda_compatlibs.sh ${os} ${ver} ${eessi_cpu_family})"
+ret=$?
+if [ $ret -ne 0 ]; then
+  echo $latest_cuda_compat_url
+  exit 1
+fi
+
+# Create a general space for our NVIDIA compat drivers
+if [ -w /cvmfs/pilot.eessi-hpc.org/host_injections ]; then
+  mkdir -p /cvmfs/pilot.eessi-hpc.org/host_injections/nvidia
+  cd /cvmfs/pilot.eessi-hpc.org/host_injections/nvidia
+else
+  echo "Cannot write to eessi host_injections space, exiting now..." >&2
+  exit 1
+fi
+
+# Check if we have any version installed by checking for the existence of /cvmfs/pilot.eessi-hpc.org/host_injections/nvidia/latest
+
+driver_cuda_version=$(nvidia-smi  -q --display=COMPUTE | grep CUDA | awk 'NF>1{print $NF}' | sed s/\\.//)
+eessi_cuda_version=$(LD_LIBRARY_PATH=/cvmfs/pilot.eessi-hpc.org/host_injections/nvidia/latest/compat/:$LD_LIBRARY_PATH nvidia-smi  -q --display=COMPUTE | grep CUDA | awk 'NF>1{print $NF}' | sed s/\\.//)
+if [ "$driver_cuda_version" -gt "$eessi_cuda_version" ]; then  echo "You need to update your CUDA compatability libraries"; fi
+
+# Check if our target CUDA is satisfied by what is installed already
+# TODO: Find required CUDA version and see if we need an update
+
+# If not, grab the latest compat library RPM or deb
+# download and unpack in temporary directory, easier cleanup after installation
+host_injections_dir=$(dirname $(realpath $0))
+tmpdir=$(mktemp -d)
+cd $tmpdir
+compat_file=${latest_cuda_compat_url##*/}
+wget ${latest_cuda_compat_url}
+
+# Unpack it
+# (the requirements here are OS dependent, can we get around that?)
+# (for rpms looks like we can use https://gitweb.gentoo.org/repo/proj/prefix.git/tree/eclass/rpm.eclass?id=d7fc8cf65c536224bace1d22c0cd85a526490a1e)
+# (deb files can be unpacked with ar and tar)
+file_extension=${compat_file##*.}
+if [[ ${file_extension} == "rpm" ]]; then
+  rpm2cpio ${compat_file} | cpio -idmv
+elif [[ ${file_extension} == "deb" ]]; then
+  ar x ${compat_file}
+  tar xf data.tar.*
+else
+  echo "File extension of cuda compat lib not supported, exiting now..." >&2
+  exit 1
+fi
+cd $host_injections_dir
+# TODO: This would prevent error messages if folder already exists, but could be problematic if only some files are missing in destination dir
+mv -n ${tmpdir}/usr/local/cuda-* .
+rm -r ${tmpdir}
+
+# Add a symlink that points to the latest version
+latest_cuda_dir=$(find . -maxdepth 1 -type d | grep -i cuda | sort | tail -n1)
+echo $latest_cuda_dir
+ln -sf ${latest_cuda_dir} latest
+
+if [ ! -e latest ] ; then
+  echo "Symlink to latest cuda compat lib version is broken, exiting now..."
+  exit 1
+fi
+
+# Create the space to host the libraries
+mkdir -p /cvmfs/pilot.eessi-hpc.org/host_injections/${EESSI_PILOT_VERSION}/compat/${os_family}/${eessi_cpu_family}
+# Symlink in the path to the latest libraries
+if [ ! -d "/cvmfs/pilot.eessi-hpc.org/host_injections/${EESSI_PILOT_VERSION}/compat/${os_family}/${eessi_cpu_family}/lib" ]; then
```

Comment:
What should we do if it _does_ exist? We should probably check that it is the symlink we expected and error if it is not.

Reply:
`host_injections_dir` is `/cvmfs/pilot.eessi-hpc.org/host_injections/nvidia` so I would set that early and use it where possible (like around L95).

Similarly, `/cvmfs/pilot.eessi-hpc.org/host_injections/${EESSI_PILOT_VERSION}/compat/${os_family}/${eessi_cpu_family}` is used in a few places, I would have that as a variable as early as possible and then reuse that.